examples/basic_scan_async/main.go

```diff
@@ -17,37 +17,34 @@ func main() {
 	// with a 5 minute timeout.
 	s, err := nmap.NewScanner(
 		nmap.WithTargets("google.com", "facebook.com", "youtube.com"),
-		nmap.WithPorts("80,443,843"),
+		nmap.WithPorts("80,443,843,1-100"),
+		nmap.WithStatsEvery("1s"),
 	)
 	if err != nil {
 		log.Fatalf("unable to create nmap scanner: %v", err)
 	}
 
 	// Executes asynchronously, allowing results to be streamed in real time.
-	if err := s.RunAsync(); err != nil {
+	nmapResults, nmapErrors, err := s.RunAsync()
+	if err != nil {
 		panic(err)
 	}
 
-	// Connect to stdout of scanner.
-	stdout := s.GetStdout()
-
-	// Connect to stderr of scanner.
-	stderr := s.GetStderr()
-
 	// Goroutine to watch for stdout and print to screen. Additionally it stores
-	// the bytes intoa variable for processiing later.
+	// the bytes into a variable for processing later.
 	go func() {
-		for stdout.Scan() {
-			fmt.Println(stdout.Text())
-			resultBytes = append(resultBytes, stdout.Bytes()...)
+		for result := range nmapResults {
+			s.Progress(result)
+			//fmt.Print(string(result))
```

Comment:
This commented-out debug line should be removed when the draft moves becomes a PR! :)